src/tools/pkcs11-tool.c

```diff
@@ -6556,9 +6556,31 @@ static int read_object(CK_SESSION_HANDLE session)
 						ASN1_PRINTABLESTRING_free(curve);
 					} else if (d2i_ASN1_OBJECT(&obj, &a, (long)len) != NULL) {
 						int nid = OBJ_obj2nid(obj);
-						if (nid != NID_ED25519 && nid != NID_X25519) {
-							util_fatal("Unknown curve OID, expected NID_ED25519 (%d), got %d",
-									NID_ED25519, nid);
+						if (nid != NID_ED25519 &&
+								nid != NID_X25519
+#if defined(EVP_PKEY_ED448)
+								&& nid != NID_ED448
+#endif
+#if defined(EVP_PKEY_X448)
+								&& nid != NID_X448
+#endif
```

Comment:
Do we know about some implementation that would have only one of these which would prevent us writing something like this?
```suggestion
#if defined(EVP_PKEY_ED448) && defined(EVP_PKEY_X448)
								&& nid != NID_ED448 && nid != NID_X448
#endif
```

Reply:
I can't tell you, I am only use OpenSSL v1.1.1 and OpenSSL v3.
Please, let me know if I take your suggested changes.